core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -408,18 +414,107 @@ public int registerOrForward(String subject,
     }
   }
 
+  @Override
+  public void deleteSchemaVersion(String subject,
+                                  Schema schema)
+      throws SchemaRegistryException {
+    try {
+      // Ensure cache is up-to-date before any potential writes
+      kafkaStore.waitUntilKafkaReaderReachesLastOffset(kafkaStoreTimeoutMs);
+      SchemaValue schemaValue = new SchemaValue(schema);
+      schemaValue.setDeleted(true);
+      kafkaStore.put(new SchemaKey(subject, schema.getVersion()), schemaValue);
+      if (!getAllVersions(subject, false).hasNext() && getCompatibilityLevel(subject) != null) {
+        deleteSubjectCompatibility(subject);
+      }
+    } catch (StoreTimeoutException te) {
+      throw new SchemaRegistryTimeoutException("Write to the Kafka store timed out while", te);
+    } catch (StoreException e) {
+      throw new SchemaRegistryStoreException("Error while deleting the schema in the"
+                                             + " backend Kafka store", e);
+    }
+  }
+
+
+  public void deleteSchemaVersionOrForward(String subject,
+                                           Schema schema)
+      throws SchemaRegistryException {
+
+    synchronized (masterLock) {
+      if (isMaster()) {
+        deleteSchemaVersion(subject, schema);
+      } else {
+        // forward registering request to the master
+        if (masterIdentity != null) {
+          forwardDeleteSchemaVersionRequestToMaster(subject, schema.getVersion());
+        } else {
+          throw new UnknownMasterException("Register schema request failed since master is "
+                                           + "unknown");
+        }
+      }
+    }
+  }
+
+  @Override
+  public List<Integer> deleteSubject(String subject) throws SchemaRegistryException {
```

Comment:
I haven't checked the commit history, but do you want to keep these on a separate branch so we can incrementally add deleting a version vs deleting the entire subject?